src/misc/api_trace.cc

```diff
@@ -0,0 +1,418 @@
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommDeregister_fn,             35);
+
+
+#undef RCCL_ASSERT_OFFSET
+
+    static_assert(
+        sizeof(rcclApiFuncTable) == compute_table_size(36),
+        "Update table major/step version and add a new offset assertion if this fails to compile");
+
+    static std::array<unsigned char, sizeof(rcclApiFuncTable)> m_buffer;
+
+
+    rcclApiFuncTable* RcclGetFunctionTable_impl()
+    {
+        static rcclApiFuncTable tbl = {sizeof(rcclApiFuncTable),
+                                        &ncclAllGather_impl,
+                                        &ncclAllReduce_impl,
+                                        &ncclAllToAll_impl,
+                                        &ncclAllToAllv_impl,
+                                        &ncclBroadcast_impl,
+                                        &ncclGather_impl,
+                                        &ncclReduce_impl,
+                                        &ncclReduceScatter_impl,
+                                        &ncclScatter_impl,
+                                        &ncclSend_impl,
+                                        &ncclRecv_impl,
+                                        &ncclRedOpCreatePreMulSum_impl,
+                                        &ncclRedOpDestroy_impl,
+                                        &ncclGroupStart_impl,
+                                        &ncclGroupEnd_impl,
+                                        &ncclGetVersion_impl,
+                                        &ncclGetUniqueId_impl,
+                                        &ncclCommInitRank_impl,
+                                        &ncclCommInitAll_impl,
+                                        &ncclCommInitRankConfig_impl,
+                                        &ncclCommFinalize_impl,
+                                        &ncclCommDestroy_impl,
+                                        &ncclCommAbort_impl,
+                                        &ncclCommSplit_impl,
+                                        &ncclGetErrorString_impl,
+                                        &ncclGetLastError_impl,
+                                        &ncclCommGetAsyncError_impl,
+                                        &ncclCommCount_impl,
+                                        &ncclCommCuDevice_impl,
+                                        &ncclCommUserRank_impl,
+                                        &ncclMemAlloc_impl,
+                                        &mscclLoadAlgo_impl,
+                                        &mscclRunAlgo_impl,
+                                        &mscclUnloadAlgo_impl,
+                                        &ncclCommRegister_impl,
+                                        &ncclCommDeregister_impl};
+
+        #if defined(RCCL_ROCPROFILER_REGISTER) && RCCL_ROCPROFILER_REGISTER > 0
+            std::array<void*, 1> table_array{&tbl};
```

Comment:
```suggestion
            std::array<void*, 1> table_array{tbl};
```
Due to modifications above, `tbl` is now a pointer